src/nnp_force.F

```diff
@@ -0,0 +1,682 @@
+         std = SUM((SUM(nnp%atomic_energy, 1) - nnp%nnp_potential_energy)**2)
+         std = std/REAL(nnp%n_committee, dp)
+         std = SQRT(std)
+         WRITE (unit_nr, fmt_string) nnp%nnp_potential_energy, std, SUM(nnp%atomic_energy, 1)
+      END IF
+
+   END SUBROUTINE nnp_print_energies
+
+! **************************************************************************************************
+!> \brief Print nnp forces
+!> \param nnp ...
+!> \param print_key ...
+!> \date   2020-10-10
+!> \author Christoph Schran (christoph.schran@rub.de)
+! **************************************************************************************************
+   SUBROUTINE nnp_print_forces(nnp, print_key)
+      TYPE(nnp_type), POINTER                            :: nnp
+      TYPE(section_vals_type), POINTER                   :: print_key
+
+      CHARACTER(len=*), PARAMETER :: routineN = 'nnp_print_forces', &
+         routineP = moduleN//':'//routineN
+
+      CHARACTER(len=default_path_length)                 :: fmt_string, middle_name
+      INTEGER                                            :: i, j, unit_nr
+      TYPE(cp_logger_type), POINTER                      :: logger
+
+      NULLIFY (logger)
+      logger => cp_get_default_logger()
+
+      ! TODO use write_particle_coordinates from particle_methods.F
+      DO i = 1, nnp%n_committee
+         WRITE (fmt_string, *) i
+         WRITE (middle_name, "(A,A)") "nnp-forces-", ADJUSTL(TRIM(fmt_string))
+         unit_nr = cp_print_key_unit_nr(logger, print_key, extension=".xyz", &
+                                        middle_name=TRIM(middle_name))
+         IF (unit_nr > 0) THEN
+            WRITE (unit_nr, *) nnp%num_atoms
+            WRITE (unit_nr, "(A,1X,A,A,F20.9)") "NNP forces [a.u.] of committee member", ADJUSTL(TRIM(fmt_string)), &
+               "energy [a.u.]=", nnp%committee_energy(i)
+
+            fmt_string = "(A4,1X,3F20.10)"
+            DO j = 1, nnp%num_atoms
+               WRITE (unit_nr, fmt_string) nnp%atoms(nnp%sort_inv(j)), nnp%committee_forces(:, j, i)
+            END DO
+         END IF
+         CALL cp_print_key_finished_output(unit_nr, logger, print_key)
+      END DO
+
+   END SUBROUTINE nnp_print_forces
+
+! **************************************************************************************************
+!> \brief Print standard deviation sigma of NNP forces
+!> \param nnp ...
+!> \param unit_nr ...
+!> \date   2020-10-10
+!> \author Christoph Schran (christoph.schran@rub.de)
+! **************************************************************************************************
+   SUBROUTINE nnp_print_force_sigma(nnp, unit_nr)
+      TYPE(nnp_type), POINTER                            :: nnp
+      INTEGER, INTENT(IN)                                :: unit_nr
+
+      CHARACTER(len=*), PARAMETER :: routineN = 'nnp_print_force_sigma', &
+         routineP = moduleN//':'//routineN
+
+      CHARACTER(len=default_path_length)                 :: fmt_string
```

Comment:
`PARAMETER`?

Reply:
I assume you mean that `routineN` is obsolete, if I'm not using it for anything? I will remove those where ever this is the case.

Reply:
no, sorry, I mean that `fmt_string` here could be made a `PARAMETER` (with `len=*`) since you write it only once

Reply:
Since `fmt_string` is used only once, it could just be inlined.